dace/sdfg/validation.py

```diff
@@ -55,6 +58,46 @@ def validate_sdfg(sdfg: 'dace.sdfg.SDFG'):
                     "storage type. Please use a different storage location." %
                     name, sdfg, None)
 
+            #Check for valid bank assignments
+            try:
+                bankassignment = sdutil.parse_location_bank(desc)
+            except ValueError as e:
+                raise InvalidSDFGError(
+                    f"Failed to parse location['bank'] for "
+                    f" array {name}: {str(e)}", sdfg, None)
+            if bankassignment is not None:
+                if bankassignment[0] == "DDR":
+                    try:
+                        tmp = int(bankassignment[1])
+                    except ValueError:
+                        raise InvalidSDFGError(
+                            "bank assignment must be convertible to int"
+                            f" for array {name}", sdfg, None)
```

Comment:
Just let the error propagate

Reply:
Are you sure that this is what we want here? The reason why it is converted to int is just to check if it is convertible. I would get confused if the validation would throw a cannot convert tmp = int(bank_assignment) to int or so without any info why this error is happening

Reply:
You're right, but then at least report the number, `"Expect integer identifier for bank assignment, got {bankassignment[1]}."`